dss/events/handlers/index.py

```diff
@@ -126,6 +128,48 @@ def create_index_data(handle: BlobStore, bucket_name: str, bundle_id: str, manif
     return index
 
 
+def get_index_shape_identifier(index_document: dict, logger) -> str:
+    """ Return string identifying the shape/structure/format of the data in the index document,
+    so that it may be indexed appropriately.
+
+    Currently, this returns a string identifying the metadata schema release major number:
+    For example:
+        v3 - Bundle contains metadata in the version 3 format
+        v4 - Bundle contains metadata in the version 4 format
+        ...
+
+    This includes verification that schema major number is the same for all index metadata
+    files in the bundle, as is the working plan for the 2017 Q4 demo.
+    If no metadata version information is contained in the bundle, the empty string is returned.
+    Currently this occurs in the case of the empty bundle used for deployment testing.
+
+    If/when bundle schemas are available, this function should be updated to reflect the
+    bundle schema type and major version number.
+
+    Other projects (non-HCA) may manage their metadata schemas (if any) and schema versions.
+    This should be an extension point that is customizable by other projects according to their metadata.
+    """
+
+    schema_version_map = defaultdict(set)  # type: typing.MutableMapping[str, typing.MutableSet[str]]
+    files = index_document['files']
+    for filename, file_content in files.items():
+        core = file_content.get('core')
+        if core is not None:
+            schema_type = core['type']
+            schema_version = core['schema_version']
+            schema_version_major = schema_version.split(".")[0]
+            schema_version_map[schema_version_major].add(schema_type)
+        else:
+            logger.info("%s", (f"File {filename} does not contain a 'core' section to identify "
+                               "the schema and schema version."))
+    if schema_version_map:
+        assert len(schema_version_map.keys()) == 1, \
+            "The bundle contains mixed schema major version numbers: {}".format(sorted(list(schema_version_map.keys())))
+        return "v" + list(schema_version_map.keys())[0]
+    else:
+        return ""  # No files with schema identifiers were found
```

Comment:
I changed my mind: I think this should be `None`. That is IMO the most natural value for something that "you don't have" or "does not exist". 

get_es_index_name should be changed to explicitly test for None rather than any false-ish value.